cli/src/commands/router/commands/compose.ts

```diff
@@ -121,6 +131,113 @@ export default (opts: BaseCommandOptions) => {
       }),
     });
 
+    routerConfig.version = createHash('sha1').update(routerConfig.toJsonString()).digest('hex');
+
+    if (config.feature_flags && config.feature_flags.length > 0) {
+      const ffConfigs: FeatureFlagRouterExecutionConfigs = new FeatureFlagRouterExecutionConfigs();
+
+      // @TODO This logic should exist only once in the shared package and reused across
+      // control-plane and cli
+
+      for (const ff of config.feature_flags) {
+        const subgraphs: Subgraph[] = [];
+
+        // Replace base subgraphs with feature flag subgraphs
+        for (const s of config.subgraphs) {
+          const featureSubgraph = ff.feature_graphs.find((ffs) => ffs.subgraph_name === s.name);
+          if (featureSubgraph) {
+            if (featureSubgraph?.schema?.file) {
+              const schemaFile = resolve(inputFileLocation, featureSubgraph.schema.file);
+              const sdl = (await readFile(schemaFile)).toString();
+              subgraphSDLs.set(featureSubgraph.name, sdl);
+            } else {
+              const result = await introspectSubgraph({
+                subgraphURL: featureSubgraph?.introspection?.url ?? featureSubgraph.routing_url,
+                additionalHeaders: Object.entries(featureSubgraph.introspection?.headers ?? {}).map(([key, value]) => ({
+                  key,
+                  value,
+                })),
+                rawIntrospection: featureSubgraph.introspection?.raw,
+              });
+
+              if (!result.success) {
+                program.error(
+                  `Could not introspect feature-graph subgraph ${featureSubgraph.name}: ${
+                    result.errorMessage ?? 'failed'
+                  }`,
+                );
+              }
+            }
+
+            subgraphs.push({
+              name: featureSubgraph.name,
+              routing_url: featureSubgraph.routing_url,
+              schema: featureSubgraph.schema,
+              subscription: featureSubgraph.subscription,
+              introspection: featureSubgraph.introspection,
+            });
+          } else {
+            subgraphs.push(s);
+          }
+        }
+
+        const result = composeSubgraphs(
+          subgraphs.map((s, index) => ({
+            name: s.name,
+            url: normalizeURL(s.routing_url),
+            definitions: parse(subgraphSDLs.get(s.name) ?? ''),
+          })),
+        );
+
+        if (result.errors && result.errors.length > 0) {
+          program.error(`Failed to compose for feature flags: ${result.errors[0]}`);
+        }
+
+        if (!result.federationResult) {
+          program.error('Failed to compose given subgraphs for feature flags');
+        }
+
+        const federatedClientSDL = result.federationResult.shouldIncludeClientSchema
+          ? printSchemaWithDirectives(result.federationResult.federatedGraphClientSchema)
```

Comment:
No, this is correct.

Reply:
Whats the difference to this line
https://github.com/wundergraph/cosmo/pull/853/files#diff-72761a9584e643f9d804e24a54b0f757dbdb07105497d247e36935a2fe9c0520R108

Reply:
One of the lines are wrong. :D

Reply:
@StarpTech everywhere else we use `printSchema` for the client schema. Please could you change this line (this line only) to `printSchema` rather than `printSchemaWithDirectives`?

Reply:
Done